packages/flutter/test/material/text_field_test.dart

```diff
@@ -3382,4 +3382,175 @@ void main() {
+      // is only one line, the first line should be overriden by the
+      // actual height.
+      //
+      // TODO(garyq): Not a round number because of rounding in LibTxt height
+      // metrics. This rounding is slated to be removed as it is no longer
+      // needed. This test should be fixed when the rounding is removed.
+      closeTo(40.00000023841858, 0.000001),
+    );
+
+    await tester.enterText(find.byType(TextField), '一\n');
+    await tester.pump();
+    expect(
+      tester.getSize(find.byType(TextField)).height,
+      // TODO(garyq): Not a round number because of rounding in LibTxt height
+      // metrics. This rounding is slated to be removed as it is no longer
+      // needed. This test should be fixed when the rounding is removed.
+      closeTo(40.00000023841858, 0.000001),
+    );
+
+    await tester.enterText(find.byType(TextField), '一\n二\n三');
+    await tester.pump();
+    expect(
+      tester.getSize(find.byType(TextField)).height,
+      // TODO(garyq): Not a round number because of rounding in LibTxt height
+      // metrics. This rounding is slated to be removed as it is no longer
+      // needed. This test should be fixed when the rounding is removed.
+      closeTo(40.00000023841858, 0.000001),
+    );
+
+    await tester.enterText(find.byType(TextField), '一\n二\n三\n四');
+    await tester.pump();
+    expect(
+      tester.getSize(find.byType(TextField)).height,
+      // TODO(garyq): Not a round number because of rounding in LibTxt height
+      // metrics. This rounding is slated to be removed as it is no longer
+      // needed. This test should be fixed when the rounding is removed.
+      closeTo(40.00000023841858, 0.000001),
+    );
+  });
+
+  testWidgets('caret size and position', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      overlay(
+        child: Container(
+          width: 300.0,
+          child: const TextField(
+            maxLines: 5,
+          ),
+        ),
+      ),
+    );
+
+    final RenderEditable editable = findRenderEditable(tester);
+    await tester.enterText(find.byType(TextField), 'abcd\n一二三四\n一b三d');
+    await tester.pump();
+
+    Rect caretRect = editable.getLocalRectForCaret(const TextPosition(offset: 2));
+    // TODO(garyq): Not a round number because of rounding in LibTxt height
+    // metrics. This rounding is slated to be removed as it is no longer
+    // needed. This test should be fixed when the rounding is removed.
+    expect(caretRect.height, closeTo(16.00000023841858, 0.000001));
+    expect(caretRect.topLeft.dx, 32);
+    expect(caretRect.topLeft.dy, closeTo(-2.384185791015625e-7, 0.000001));
+
+    caretRect = editable.getLocalRectForCaret(const TextPosition(offset: 4));
+    // TODO(garyq): Not a round number because of rounding in LibTxt height
+    // metrics. This rounding is slated to be removed as it is no longer
+    // needed. This test should be fixed when the rounding is removed.
```

Comment:
please include a link to the relevant issue, so that audits of issue links will reveal this